lib/tag-finder.coffee

```diff
@@ -8,14 +8,14 @@ SelfClosingTags = require './self-closing-tags'
 module.exports =
 class TagFinder
   constructor: (@editor) ->
-    @tagPattern = /(<(\/?))([^\s>]+)([\s>]|$)/
-    @wordRegex = /[^>\r\n]*/
+    @tagPattern = /(<(\/?))([^\s>]+)([^/>]*)(\/?)(>)/
+    @wordRegex = /[^>\r\n]*>/
```

Comment:
So the issue that I'm trying to fix is that "custom" self-closing tags, like a self-closing `span` element would not get identified properly by the `tag-finder.coffee` script and get treated the same as opening tags. I realized this was because the `@tagPattern` RegEx that found the tags only matched tags up to the `space` right before the attributes (class, id, style, etc.).

In other words the `tag-finder` was not going all the way past the attributes to the end of the tag, the closing angle bracket `>`, when identifying tags.

However, going all the way past the attributes to see whether the tag ends in `/>` or just a `>` was essential to determining whether or not the tag closes itself, and based on that, whether or not it should be matched with other closing tags.